client/src/features/visualizer-vivagraph/store/tangle.ts

```diff
@@ -37,6 +39,13 @@ export const useTangleStore = create<TangleState>()(
     devtools((set, get) => ({
         ...INITIAL_STATE,
 
+        resetTangleStore: () => {
+            set(() => {
+                return {
+                    ...INITIAL_STATE,
+                };
+            });
+        },
```

Comment:
This is not really what you expect, because at the start, when creating the store you are actually passing the reference to the Map stored in the INITIAL_STATE, so instead just manually copy the INITIAL_STATE so it doesn't use references to previous maps. You can actually check this by setting a console log in like this:

```typescript
        resetTangleStore: () => {
            set(() => {
                return {
                    ...INITIAL_STATE,
                };
            });
            console.log("after cleanup", get().blockIdToMetadata);
        },
```

Reply:
```suggestion
        resetTangleStore: () => {
            set(() => {
                return {
                        blockIdToMetadata: new Map(),
                        visibleBlocks: [],
                        selectedNode: null,
                        search: "",
                };
            });
        },
```